rel/value_set_test.go

```diff
@@ -39,3 +25,36 @@ func TestAsString(t *testing.T) {
 	// require.True(t, isString)
 	// assert.True(t, stringified.Equal(NewString([]rune(""))))
 }
+
+func TestGenericSetCallAll(t *testing.T) {
+	t.Parallel()
+
+	set := NewSet(
+		NewTuple(
+			NewAttr("@", NewNumber(1)),
+			NewAttr("@fooo", NewNumber(42)),
+		),
+		NewTuple(
+			NewAttr("@", NewNumber(1)),
+			NewAttr("@baar", NewNumber(24)),
+		),
+		NewTuple(
+			NewAttr("@", NewNumber(2)),
+			NewAttr("@foo", NewNumber(3)),
+			NewAttr("@bar", NewNumber(4)),
+		),
+		NewTuple(
+			NewAttr("@", NewNumber(3)),
+		),
+		NewTuple(
+			NewAttr("@", NewNumber(4)),
+			NewAttr("random", NewNumber(5)),
+		),
+	)
+
+	AssertEqualValues(t, NewSet(NewNumber(42), NewNumber(24)), set.CallAll(NewNumber(1)))
+	AssertEqualValues(t, NewSet(NewNumber(3), NewNumber(4)), set.CallAll(NewNumber(2)))
+	AssertEqualValues(t, None, set.CallAll(NewNumber(3)))
```

Comment:
2 and 3 should both fail. Only binary tuples with one `@` attribute should be supported.

Reply:
they panic now